src/cli/cli.js

```diff
@@ -124,24 +125,29 @@ async function runTests (argParser) {
 }
 
 async function listBrowsers (providerName) {
-    const provider = await browserProviderPool.getProvider(providerName);
 
-    if (!provider)
-        throw new GeneralError(RUNTIME_ERRORS.browserProviderNotFound, providerName);
-
-    if (provider.isMultiBrowser) {
-        const browserNames = await provider.getBrowserList();
+    let browserNames;
 
+    try {
+        browserNames = await listBrowsersApi(providerName);
+    }
+    catch (e) {
         await browserProviderPool.dispose();
-
-        if (providerName === 'locally-installed')
-            console.log(browserNames.join('\n'));
-        else
-            console.log(browserNames.map(browserName => `"${providerName}:${browserName}"`).join('\n'));
+        throw e.code === RUNTIME_ERRORS.browserProviderNotFound
+            ? new GeneralError(e.code, providerName)
+            : e;
     }
+
+    const disposePromise = browserProviderPool.dispose();
```

Comment:
I feel that this is not enough. This only works for a CLI call. We should try to dispose the provider when `listBrowsers` is used in API. Some providers create objects (sessions, files, requests) during initialization that must be disposed to prevent data losses or leaks. But of course, we shouldn't dispose providers that are started for a test session, not for `listBrowsers`.